vendor/k8s.io/kubernetes/pkg/genericapiserver/storage_factory.go

```diff
@@ -154,7 +154,7 @@ func (s *DefaultStorageFactory) getStorageGroupResource(groupResource unversione
 
 // New finds the storage destination for the given group and resource. It will
 // return an error if the group has no storage destination configured.
-func (s *DefaultStorageFactory) New(groupResource unversioned.GroupResource) (storage.Interface, error) {
+func (s *DefaultStorageFactory) New(groupResource unversioned.GroupResource, overrideStorageVersion *unversioned.GroupVersion) (storage.Interface, error) {
```

Comment:
I didn't need to change the storageFactory at all. Feels a bit closer but I still need something I am missing. PTAL